plugins/spamassassin.js

```diff
@@ -269,10 +275,13 @@ exports.get_spamd_socket = function (next, connection, headers) {
     const results_timeout = parseInt(plugin.cfg.main.results_timeout) || 300;
 
     socket.on('connect', function () {
+        // Abort if the transaction is gone
         if (!connection.transaction) {
+            plugin.logwarn(connection, 'Transaction gone, cancelling SPAMD connection');
             socket.end();
-            return;
+            return false;  // next() is called in socket.on('end')
```

Comment:
Why is this not a code smell?  Values should not be returned unless something is checking them.